index.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+const assert = require('assert');
+
+class DejaVu {
+
+  /**
+   * Constructs an new instance of DejaVu to identify "original" events.
+   * @param {Object} options The constructor options.
+   * @param {Object} options.redisConnection A connection to a Redis deployment.
+   */
+  constructor(options) {
+    assert(options.redisConnection, 'Must provide a Redis connection');
+    this._redisConnection = options.redisConnection;
+    this._handlers = {};
+  }
+
+  /**
+   * isNew returns if an event is new or not.
+   * @param {String} prefix The prefix to namespace events under.
+   * @param {Number} window The window to filtet event timestamps by (in ms).
```

Comment:
Actually if the client specifies a `ttl` to `markAsSeen`, why do they also need to specify a time limit here? —doesn't the ttl effectively determine the time limit? Only reason to have a limit here would be if the client wanted to use different time limits per event but that seems weird, I envision them setting general rules for considering events new.